.github/scripts/generate_release_notes.py

```
@@ -0,0 +1,214 @@
+- Uses Claude API to generate release notes from git commits
+- Analyzes changes between tags/commits
+- Produces structured markdown release notes
+"""
+
+import os
+import sys
+import subprocess
+import anthropic
+
+
+def get_commits_since_tag(tag: str = None) -> str:
```

Ruff (0.14.14)

[warning] PEP 484 prohibits implicit `Optional`

Convert to `T | None`

(RUF013)